h2/src/main/org/h2/schema/Sequence.java

```diff
@@ -139,17 +138,14 @@ public synchronized void modify(Long startValue, Long minValue,
      * @param maxValue the prospective max value
      * @param increment the prospective increment
      */
-    private static boolean isValid(long value, long minValue, long maxValue,
-            long increment) {
+    private static boolean isValid(long value, long minValue, long maxValue, long increment) {
         return minValue <= value &&
             maxValue >= value &&
             maxValue > minValue &&
             increment != 0 &&
             // Math.abs(increment) < maxValue - minValue
-            // use BigInteger to avoid overflows when maxValue and minValue
-            // are really big
-            BigInteger.valueOf(increment).abs().compareTo(
-                    BigInteger.valueOf(maxValue).subtract(BigInteger.valueOf(minValue))) < 0;
+            // Can use Long.compareUnsigned() on Java 8
+            Math.abs(increment) + Long.MIN_VALUE < maxValue - minValue + Long.MIN_VALUE;
```

Comment:
`Long.MAX_VALUE - Long.MIN_VALUE == 0xffffffffffffffffL` that is greater or equal to any other unsigned long value. Unsigned arithmetic is used here, overflow is not possible.

This can be written as `Long.compareUnsigned(Math.abs(increment), maxValue - minValue) < 0`, but Java 8 is required for it.

Reply:
We know that `maxValue > minValue` due to previous checks, so result of subtraction cannot be negative and we can treat it as unsigned long. `Math.abs(long)` can be also treated as unsigned long, in this case `Math.abs(Long.MIN_VALUE)` doesn't need special handing. So the whole condition is very simple.

Reply:
Thanks.